### ISSUE TITLE
Can't close a subscription node

### ISSUE THREAD

**Issue by remy** [prisma/prisma1#3955]
When using the prisma API, there's no way to stop a `prisma.$subscribe.X().node` - either for testing subscriptions (without mocking) or for programmatic control over subscriptions in a node process.

**Expected behavior**

There should be a method to call against the `node()` response that allows me to stop the subscription.

**Versions (please complete the following information):**
 - Connector: `Postgres`
 - `prisma` CLI: `prisma/1.25.3 (darwin-x64) node-v10.4.0`
 - OS: `OS X High Sierra`
 - `prisma-binding@2.2.14`

```
backend@1.0.0 /Users/remy/Sites/deck/backend
├─┬ prisma@1.25.3
│ └─┬ prisma-cli-core@1.25.3
│   └── prisma-client-lib@1.25.3
└── prisma-client-lib@1.24.0
```

**Comment by remy** [comment]
Also note that stopping the server doesn't seem to close off the connection.

```
const app = await server.start() // where server is `new GraphQLServer`
app.close(); // leaves subscription open
```

**Comment by pantharshit00** [comment]
@timsuchanek Exposing `close()` from subscription-transport-ws` can be an option here but it adds to the API surface.

https://github.com/apollographql/subscriptions-transport-ws#close--void---closes-the-websocket-connection-manually-and-ignores-reconnect-logic-if-it-was-set-to-true

**Comment by divyenduz** [comment]
@remy Thanks for raising this issue. Prisma client returns an [async iterator](https://tc39.github.io/proposal-async-iteration/) that exposes a `return` method to perform cleanups. 

I create a [small demo here](https://github.com/divyenduz/prisma-3955) to show that the async iterator generated by the Prisma client subscription does indeed expose and respect the return method. 

It is the responsibility of the caller of this async iterator to call return when the subscription is stopped, in this case the GraphQL server. 

Do you have a reproduction for a specific GraphQL server that leaks subscriptions?

**Comment by remy** [comment]
In my full complex app I do, but otherwise no(!)

I've had to give up using subscriptions (I switched to using the alt await
method some time ago). It would frequently miss events and it's needed for
user sign up emails (when their status changes).

I did find that combining UPDATE and CREATE event listening on a
subscription for a particular field completely fails though. It never fires
the create event. You might be able to replicate in a small repo.
